File: web/src/services/tracker/ProjectService.js
import { request, METHOD } from "@/utils/request";

export function findProjectCategories() {
  return request({
    url: "/tracker/project/categories",
    method: METHOD.GET,
  });
}

export function findProjects(parameter, filter) {
  return request({
    url: "/tracker/project/list",
    method: METHOD.POST,
    params: parameter,
    data: filter,
  });
}
export function findOneProject(id) {
  return request({
    url: `/tracker/project/${id}`,
    method: METHOD.GET,
  });
}
export function findPermissions() {
  return request({
    url: "/tracker/project/permissions",
    method: METHOD.GET,
  });
}
export function createProject(parameter) {
  return request({
    url: "/tracker/project",
    method: METHOD.POST,
    data: parameter,
  });
}

export function updateProject(parameter) {
  return request({
    url: "/tracker/project",
    method: METHOD.PUT,
    data: parameter,
  });
}
export function deleteProject(id) {
  return request({
    url: "tracker/project" + "/" + id,
    method: METHOD.DELETE,
  });
}
export function clearTrackerItemData(id) {
  return request({
    url: "tracker/project/clear-item/" + id,
    method: METHOD.DELETE,
  });
}
export function checkDuplicateProjectName(id, name) {
  return request({
    url: "/tracker/project/checkDuplicateProjectName",
    method: METHOD.GET,
    params: { id: id, name: name },
  });
}
export function checkDuplicateProjectShortName(id, shortName) {
  return request({
    url: "/tracker/project/checkDuplicateProjectShortName",
    method: METHOD.GET,
    params: { id: id, shortName: shortName },
  });
}
export function findNotEndedProjects(keyword) {
  return request({
    url: "tracker/project/notEndedProjects",
    method: METHOD.GET,
    params: { keyword: keyword },
  });
}
export function copyProject(parameter) {
  return request({
    url: "/tracker/project/copyProject",
    method: METHOD.POST,
    data: parameter,
  });
}
export function findRecentProjects(keyword) {
  return request({
    url: "/tracker/project/recentProjects",
    method: METHOD.GET,
    params: { keyword: keyword },
  });
}

export function recordRecentProject(projectId) {
  return request({
    url: "/tracker/project/recordProject",
    method: METHOD.GET,
    params: { projectId: projectId },
  });
}

File: server/helm-tracker/src/main/java/com/harbortek/helm/tracker/service/ProjectService.java
/*
 * Copyright [2025] [Harbortek Corp.]
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.harbortek.helm.tracker.service;

import com.harbortek.helm.tracker.entity.project.RecentProjectEntity;
import com.harbortek.helm.tracker.vo.ProjectCopyVo;
import com.harbortek.helm.tracker.vo.ProjectVo;
import com.harbortek.helm.tracker.vo.RecentProjectVo;
import com.harbortek.helm.tracker.vo.template.ProjectTemplateVo;
import com.harbortek.helm.tracker.vo.view.ObjectFilter;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.Collection;
import java.util.List;

public interface ProjectService {
//	ProjectVo createProject(ProjectVo project);

//	ProjectVo copyProject(ProjectCopyVo projectCopyVo);

	ProjectVo updateProject(ProjectVo project);

	void deleteProject(Long id);

	List<ProjectVo> findAll();

	List<ProjectVo> findByOwner(Long ownerId);

//	Collection<ProjectVo> findByWorkspace(WorkSpace workSpace);
//
//	Collection<ProjectVo> findByProjectSet(ProjectSetEntity projectSet);

	ProjectTemplateVo saveAsTemplate(ProjectVo project);

	ProjectVo findOneProject(Long id);

	Page<ProjectVo> findByCategory(Long categoryId, String keyword, String status, ObjectFilter filter, Pageable pageable);

	Boolean checkDuplicateProjectName(Long projectId, String name);

	Boolean checkDuplicateProjectShortName(Long projectId, String shortName);

	long countUnFinishedProjects(Long userId);

	void clearTrackerItemData(Long id);

	List<RecentProjectVo> findRecentProjects(Long userId);

	RecentProjectEntity recordRecentProject(Long userId, Long projectId);

}
